crates/pep508-rs/src/verbatim_url.rs

```diff
@@ -57,6 +57,9 @@ impl VerbatimUrl {
         let mut url = Url::from_file_path(path.clone())
             .map_err(|()| VerbatimUrlError::UrlConversion(path.to_path_buf()))?;
 
+        // Drop trailing slashes.
+        url.path_segments_mut().unwrap().pop_if_empty();
```

Comment:
This is sort of ugly, but otherwise, we get inconsistent comparisons between the serialized and user-provided URLs.